packages/core/content-releases/admin/src/components/CMReleasesContainer.tsx

```diff
@@ -0,0 +1,280 @@
+                      onChange={(value) => setFieldValue('releaseId', value)}
+                      value={values.releaseId}
+                    >
+                      {releases?.map((release) => (
+                        <SingleSelectOption key={release.id} value={release.id}>
+                          {release.name}
+                        </SingleSelectOption>
+                      ))}
+                    </SingleSelect>
+                  </Box>
+                  <FieldLabel>
+                    {formatMessage({
+                      id: 'content-releases.content-manager-edit-view.add-to-release.action-type-label',
+                      defaultMessage: 'What do you want to do with this entry?',
+                    })}
+                  </FieldLabel>
+                  <Flex>
+                    <ReleaseActionOptions
+                      selected={values.type}
+                      handleChange={(e) => setFieldValue('type', e.target.value)}
+                    />
+                  </Flex>
+                </Flex>
+              </ModalBody>
+              <ModalFooter
+                startActions={
+                  <Button onClick={handleClose} variant="tertiary" name="cancel">
+                    {formatMessage({
+                      id: 'content-releases.content-manager-edit-view.add-to-release.cancel-button',
+                      defaultMessage: 'Cancel',
+                    })}
+                  </Button>
+                }
+                endActions={
+                  /**
+                   * TODO: Ideally we would use isValid from Formik to disable the button, however currently it always returns true
+                   * for yup.string().required(), even when the value is falsy (including empty string)
+                   */
+                  <Button type="submit" disabled={!values.releaseId} loading={isLoading}>
+                    {formatMessage({
+                      id: 'content-releases.content-manager-edit-view.add-to-release.continue-button',
+                      defaultMessage: 'Continue',
+                    })}
+                  </Button>
+                }
+              />
+            </Form>
+          );
+        }}
+      </Formik>
+    </ModalLayout>
+  );
+};
+
+export const CMReleasesContainer = () => {
+  const [showModal, setShowModal] = React.useState(false);
+  const { formatMessage } = useIntl();
+  const {
+    isCreatingEntry,
+    allLayoutData: { contentType },
+  } = useCMEditViewDataManager();
+
+  const toggleAddActionToReleaseModal = () => setShowModal((prev) => !prev);
+
+  /**
+   * - Impossible to add entry to release before it exists
+   * - Content types without draft and publish cannot add entries to release
+   * TODO v5: All contentTypes will have draft and publish enabled
+   */
+  if (isCreatingEntry || !contentType?.options?.draftAndPublish) {
+    return null;
+  }
+
+  return (
+    <CheckPermissions permissions={PERMISSIONS.main}>
```

Comment:
This is a really interesting scenario: do we want to have this CheckPermissions at this level?

Because this means that to add an entry to a release you should have both read & add entry to a release permission. 

Also, should we check here also the publish permission on this content type to show the "Add to release" button?

Reply:
That's a good point I didn't consider, but makes sense to me. Why should you be able to add an entry to release if you aren't allowed to read it 🤔 

Not sure about the publishing. Shouldn't you be allowed to add something to a release since someone else will publish it via the release?

Reply:
I think we will have to check with Product on this one. I'll bring it up today in the DSM.